<launch>

<!-- Controllers -->

	<node pkg="focus_control" type="controller" name="controller" output="screen">
	</node>

	<node pkg="focus_control" type="diff_flat_state_feedback" name="diff_flat_state_feedback" output="screen">
	</node>

	<node pkg="focus_control" type="longitudinal_controller" name="longitudinal_controller" output="screen">
	</node>

	<node pkg="focus_control" type="lateral_controller" name="lateral_controller" output="screen">
	</node>

	<node pkg="focus_control" type="kalman.py" name="kalman" output="screen">
	</node>


<!-- Communication -->

	<node pkg="focus_serial" type="serial_transmitter" name="serial_transmitter" output="screen">
	<param name="port" value="/dev/ttyUSB1" />	
	</node>


<!-- CAN Data -->
	<node pkg="pcan" type="can_publisher" name="can_publisher" output="screen">
	</node>

<!-- GPS -->	
	<node pkg="swiftnav_piksi" type="piksi_node" name="piksi_node" output="screen">
    	<param name="port" value="/dev/ttyUSB0" />
	</node>
   
	<node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >
    	<rosparam command="delete" param="/diagnostic_aggregator" />
    	<rosparam command="load" file="$(find swiftnav_piksi)/config/swiftnav_piksi_diag.yaml" />
	</node>

<!-- Vector Nav IMU -->
<!--
	<node pkg="vectornav" type="vn200_node" name="vectornav"  output="screen" > 
	  
	  <param name="serial_port" type="string" value="/dev/ttyUSB1" />
	  <param name="serial_baud" type="int"    value="115200" />
	  
      <param name="binary_data_output_port" type="int"    value="2" />
      <param name="binary_gps_data_output_rate" type="int"    value="4" />
      <param name="binary_ins_data_output_rate" type="int"    value="20" />
      <param name="binary_imu_data_output_rate" type="int"    value="100" />
	  
	  --><!-- Type: 0 None (Polling method), 19 IMU, 20 GPS, 22 INS -->
	<!--
	  <param name="async_output_type" type="int"    value="0" />
	  <param name="async_output_rate" type="int"    value="10" />
	  
	  <param name="imu/frame_id" type="string" value="LLA" />
	  <param name="gps/frame_id" type="string" value="LLA" />
	</node>	
-->

</launch>
